### PR TITLE
sys/bit: move bit.h header to sys/

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

The bit access functions are not tied to Cortex-M CPUs, here they only provide optimizations via bit-banding.

But the functions are generally useful - so move them to an arch independent location.

### Testing procedure

Everything should still compile.

### CODE REVIEW

File: sys/include/bit.h [hunk 1]
```diff
@@ -7,11 +7,11 @@
  */
 
 /**
- * @ingroup     cpu_cortexm_common
+ * @ingroup     sys
  * @{
  *
  * @file
- * @brief       Bit access macros for Cortex-M based CPUs
+ * @brief       Bit access macros with bit-banding support for Cortex-M based CPUs
```
Comment (fjmolinas) [inline sys/include/bit.h:14]: Some documentation and macros have specific `cortex-m` refernces.

Comment (fjmolinas) [reply sys/include/bit.h:14]: So are you sure its not cortex-m specific?

Comment (benpicco) [reply sys/include/bit.h:14]: The fallback functions are just normal bit operations.
But Cortex-M can have hardware support for that, mapping individual bits to byte locations in a separate RAM region.

Comment (fjmolinas) [reply sys/include/bit.h:14]: Ah Ok, but I still think the comment at
https://github.com/RIOT-OS/RIOT/blob/f85ef6f841cbcfa247beed8eb88fb4d811d701e9/sys/include/bit.h#L35
should be fixed then.

Comment (benpicco) [reply sys/include/bit.h:14]: Is that what you had in mind?

Comment (fjmolinas) [reply sys/include/bit.h:14]: yep :)

**Review by fjmolinas — APPROVED** [review]
ACK

**Pull request merged**